api-gateway/src/main/resources/static/scripts/inventory-product-list/inventory-product-list.controller.js

```diff
@@ -4,28 +4,19 @@ angular.module('inventoryProductList')
     .controller('InventoryProductController', ['$http', '$scope', '$stateParams','$window', 'InventoryService', function ($http, $scope, $stateParams, $window, InventoryService) {
         var self = this;
         var inventoryId
-
-        $http.get('api/gateway/inventory/' + $stateParams.inventoryId + '/products').then(function (resp) {
-            self.inventoryProductList = resp.data;
-            inventoryId = $stateParams.inventoryId;
-            InventoryService.setInventoryId(inventoryId);
-            if (resp.data.length === 0) {
-                // Handle if inventory is empty
-                console.log("The inventory is empty!");
-            }
-
-
-        }).catch(function (error) {
-            if (error.status === 404) {
-                console.clear()
-                console.log("State params: " + $stateParams)
-                console.log("Inventory is now empty.")
-            } else {
-                console.error('An error occurred:', error);
-            }
-        });
-
-
+        const pageSize = 15;
+        self.currentPage = $stateParams.page || 0;
+        self.pageSize = $stateParams.size || pageSize;
+        self.actualCurrentPageShown = parseInt(self.currentPage) + 1;
+        self.baseUrl = "api/gateway/inventory/" + $stateParams.inventoryId + "/products-pagination?page=" + self.currentPage + "&size=" + self.pageSize;
+        self.baseURLforTotalNumberOfProductsByFiltering = "api/gateway/inventory/" + $stateParams.inventoryId + "/products-count";
+        self.lastParams = {
+            productName: '',
+            productQuantity: '',
+            productPrice: '',
+            productSalePrice: ''
+        }
+        fetchProductList();
```

Comment:
Do you mind explaining this particular set of code?

Reply:
This is so the moment the controller gets called it fetched the product list